# HG changeset patch
# Parent 39123f891f8f51969097c7a9bbfb15db8ad11917
# User sbr
# Date 1346486603 -7200

Ctrl+Click in sign list to edit sign. (r24502)

diff -r 39123f891f8f src/lang/english.txt
--- a/src/lang/english.txt	Thu Aug 30 17:45:09 2012 +0000
+++ b/src/lang/english.txt	Sat Sep 01 10:34:12 2012 +0200
@@ -2749,6 +2749,7 @@
 STR_SIGN_LIST_CLEAR                                             :{BLACK}Clear filter
 STR_SIGN_LIST_MATCH_CASE_TOOLTIP                                :{BLACK}Toggle matching case when comparing sign names against the filter string
 STR_SIGN_LIST_CLEAR_TOOLTIP                                     :{BLACK}Clear filter string
+STR_SIGN_LIST_CLICK_ON_LIST_TOOLTIP                             :{BLACK}Click on sign to centre main view on sign position. Ctrl+Click to edit the sign text
 
 # Sign window
 STR_EDIT_SIGN_CAPTION                                           :{WHITE}Edit sign text
diff -r 39123f891f8f src/signs_gui.cpp
--- a/src/signs_gui.cpp	Thu Aug 30 17:45:09 2012 +0000
+++ b/src/signs_gui.cpp	Sat Sep 01 10:34:12 2012 +0200
@@ -252,7 +252,11 @@
 				if (id_v == INT_MAX) return;
 
 				const Sign *si = this->signs[id_v];
-				ScrollMainWindowToTile(TileVirtXY(si->x, si->y));
+				if (_ctrl_pressed) {
+					ShowRenameSignWindow(si); // Ctrl+Click to edit the sign.
+				} else {
+					ScrollMainWindowToTile(TileVirtXY(si->x, si->y)); // Click to move to sign position.
+				}
 				break;
 			}
 			case WID_SIL_FILTER_CLEAR_BTN:
@@ -394,6 +398,7 @@
 	NWidget(NWID_HORIZONTAL),
 		NWidget(NWID_VERTICAL),
 			NWidget(WWT_PANEL, COLOUR_GREY, WID_SIL_LIST), SetMinimalSize(WD_FRAMETEXT_LEFT + 16 + 255 + WD_FRAMETEXT_RIGHT, 50),
+								SetDataTip(STR_NULL, STR_SIGN_LIST_CLICK_ON_LIST_TOOLTIP),
 								SetResize(1, 10), SetFill(1, 0), SetScrollbar(WID_SIL_SCROLLBAR), EndContainer(),
 			NWidget(NWID_HORIZONTAL),
 				NWidget(WWT_PANEL, COLOUR_GREY), SetFill(1, 1),
